src/main/java/org/kapott/hbci/callback/HBCICallbackSwing.java

```diff
@@ -290,7 +290,7 @@ private void needSecret(final Hashtable<String, Object> currentData,final String
             mainbox.add(Box.createVerticalStrut(8));
             
             JPasswordField tempinput=null;
-            if (((Integer)currentData.get("reason")).intValue()==NEED_PASSPHRASE_SAVE) {
+            if (currentData.get("reason")==Reason.NEED_PASSPHRASE_SAVE) {
```

Comment:
Cast auf Integer nicht mehr notwendig, da in [Zeile 120](https://github.com/hbci4j/hbci4java/pull/60/files#diff-e2247af97d6524bbbb70ed1e2980aadd2a2db48881648a303005e9a503951387L120) der Enum-Wert hinterlegt wird.